File: backend/app/schemas/risk.py
from pydantic import BaseModel


class RiskFactor(BaseModel):
    name: str
    score: float  # 0-100 sub-score
    weight: float
    points: float | None = None  # normalized contribution to the final score
    description: str


class RiskOut(BaseModel):
    ticker: str
    risk_score: float
    risk_level: str  # low / medium / high / very_high
    factors: list[RiskFactor]
    as_of: str
    note: str

File: frontend/types/risk.ts
// Mirrors backend/app/schemas/risk.py

export type RiskLevel = "low" | "medium" | "high" | "very_high";

export interface RiskFactor {
  name: string;
  score: number; // 0-100 sub-score
  weight: number;
  points: number | null; // contribution to final score
  description: string;
}

export interface Risk {
  ticker: string;
  risk_score: number;
  risk_level: RiskLevel;
  factors: RiskFactor[];
  as_of: string;
  note: string;
}
